### PR TITLE
Tablelizer

### PR DESCRIPTION
**Pull request by christopher-schroeder**
This subcommand generates tables (tsv) from vcf files using an expression.

### CODE REVIEW

**Comment by tedil** [comment]
TODO:

- [ ] write tests
- [ ] discuss naming
- [ ] discuss subcommand vs different entrypoint (i.e. `vembrane filter` and `vembrane table` vs `vembrane` and `vcftable`)

Questions:
- Does this only use fields that appear in the expression or is there a "dump everything" mode as well?

File: vembrane/modules/tablelize.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+
+def add_subcommmand(subparsers):
+    parser = subparsers.add_parser("tablelize")
+    parser.add_argument(
+        "expression",
+        type=check_filter_expression,
+        help="Filter variants and annotations. If this removes all annotations, "
+        "the variant is removed as well.",
+    )
```
Comment (tedil) [inline vembrane/modules/tablelize.py:19]: The help string needs updating ;)

File: .flake8 [hunk 2]
```diff
@@ -1,2 +1,9 @@
 [flake8]
 max-line-length = 88
+ignore =
+    # E731: do not assign a lambda expression, use a def
+    E731,
+    # E203: whitespace before : (refs: https://github.com/PyCQA/pycodestyle/issues/373)
+    E203,
+    # W503: line break occurred before a binary operator
+    W503
```
Comment (tedil) [inline .flake8:9]: why?

File: vembrane/ann_types.py [hunk 3]
```diff
@@ -23,6 +23,9 @@ def __eq__(self, other):
     def __bool__(self):
         return False
 
+    def __repr__(self):
+        return "."
```
Comment (tedil) [inline vembrane/ann_types.py:27]: Should it really be `"."`? I'd rather have an empty cell or `None` or `NaN` I guess.

**Review by tedil — CHANGES_REQUESTED** [review]

**Review by tedil — CHANGES_REQUESTED** [review]
Almost there! Some small things that need to be addressed and then we're good to go

File: vembrane/ann_types.py [hunk 4]
```diff
@@ -23,6 +23,9 @@ def __eq__(self, other):
     def __bool__(self):
         return False
 
+    def __repr__(self):
+        return ""
+
```
[the remark this answers is not in the document]
Comment (christopher-schroeder) [reply vembrane/ann_types.py]: yes

**Review by tedil — APPROVED** [review]

**Pull request merged**